components/sup/src/manager/service/hooks.rs

```diff
@@ -66,33 +66,64 @@ pub trait Hook: fmt::Debug + Sized {
 
     fn file_name() -> &'static str;
 
+    fn deprecated_file_name() -> Option<&'static str> {
+        None
+    }
+
+    /// Tries to load a hook if a (deprecated) hook file exists.
+    ///
+    /// Returns the hook if template file (deprecated or not) is found
     fn load<C, T>(service_group: &ServiceGroup, concrete_path: C, template_path: T) -> Option<Self>
     where
         C: AsRef<Path>,
         T: AsRef<Path>,
     {
         let concrete = concrete_path.as_ref().join(Self::file_name());
         let template = template_path.as_ref().join(Self::file_name());
-        match std::fs::metadata(&template) {
-            Ok(_) => {
-                let pair = match RenderPair::new(concrete, &template) {
-                    Ok(pair) => pair,
-                    Err(err) => {
-                        outputln!(preamble service_group, "Failed to load hook: {}", err);
-                        return None;
-                    }
-                };
-                Some(Self::new(service_group, pair))
-            }
-            Err(_) => {
-                debug!(
-                    "{} not found at {}, not loading",
-                    Self::file_name(),
-                    template.display()
+        let deprecated_template =
+            Self::deprecated_file_name().map(|file_name| template_path.as_ref().join(file_name));
+
+        let has_template = std::fs::metadata(&template).is_ok();
+        let has_deprecated_template = deprecated_template
+            .and_then(|t| std::fs::metadata(&t).ok())
+            .is_some();
+
+        let template_to_use = if has_deprecated_template && !has_template {
+            warn!(
+                "Deprecated hook file detected: {}. You should use {} instead.",
+                Self::deprecated_file_name().unwrap(),
+                Self::file_name()
+            );
+            Self::deprecated_file_name().map(|file_name| template_path.as_ref().join(file_name))
+        } else if has_template {
+            if has_deprecated_template {
+                warn!(
+                    "Deprecated hook file detected along with expected one. \
+                    You should remove {} and keep only {}.",
+                    Self::deprecated_file_name().unwrap(),
+                    Self::file_name()
                 );
-                None
             }
-        }
+            Some(template)
+        } else {
+            debug!(
+                "{} not found at {}, not loading",
+                Self::file_name(),
+                template.display()
+            );
+            None
+        };
+
+        template_to_use.and_then(|t| {
+            let pair = match RenderPair::new(concrete, &t) {
+                Ok(pair) => pair,
+                Err(err) => {
+                    outputln!(preamble service_group, "Failed to load hook: {}", err);
+                    return None;
+                }
+            };
+            Some(Self::new(service_group, pair))
+        })
```

Comment:
I think this could be made a bit simpler if instead of making `template_to_use` an `Option<PathBuf>`, it was a `PathBuf`. That can be achieved by changing the code above to:
```rust
        let template_to_use = if has_template {
            if has_deprecated_template {
                …
            }
            template
        } else if has_deprecated_template {
            …
            deprecated_template.unwrap()
        } else {
            …
            return None;
        };
```
In this case the `unwrap` is safe because `has_deprecated_template` implies that `deprecated_template` is a `Some` variant. With that change, this last block becomes simpler:
```rust
        match RenderPair::new(concrete, &template_to_use) {
            Ok(pair) => Some(Self::new(service_group, pair)),
            Err(err) => {
                outputln!(preamble service_group, "Failed to load hook: {}", err);
                return None;
            }
        }
```